tracer/src/Datadog.Trace/RemoteConfigurationManagement/RemoteConfigurationCache.cs

```diff
@@ -24,5 +25,21 @@ public RemoteConfigurationCache(RemoteConfigurationPath path, int length, Dictio
         public Dictionary<string, string> Hashes { get; }
 
         public int Version { get; }
+
+        public uint ApplyState { get; private set; } = ApplyStates.UNACKNOWLEDGED;
+
+        public string Error { get; private set; }
+
+        public void Applied()
```

Comment:
This could cause potential problems if we have more than one subscriber, and first calls `ErrorOccurred` and then the second calls `Applied` - the first error would be overwritten and we'll never know about it. 

I don't think we necessarily need to deal with this right now (e.g. by refactoring away the usage of events entirely as discussed offline), since we currently only have one subscriber per `Product`, but maybe in the meantime we should consider doing a hack such as https://stackoverflow.com/a/2078459 to ensure we won't stumble onto this problem in the future?

Reply:
We have some defensive code in `ProductConfigChangedEventArgs` that stops errors being overwritten by an ack. I think that's enough for now, but clearly this is part of the system that should be rethought in the longer term.